### PR TITLE
Minor improvements to resource cleanup and copying found by FindBugs …

### PR DESCRIPTION
**Pull request by bretthoerner**
…and Infer.

Happy to say neither tool found anything critical, figured I may as well cleanup what they found though.

---
This change is [Reviewable](https://reviewable.io/reviews/getsentry/raven-java/245)

### CODE REVIEW

File: raven/src/main/java/com/getsentry/raven/event/BreadcrumbBuilder.java [hunk 1]
```diff
@@ -33,7 +33,7 @@ public BreadcrumbBuilder setType(String newType) {
      * @return current BreadcrumbBuilder
      */
     public BreadcrumbBuilder setTimestamp(Date newTimestamp) {
-        this.timestamp = newTimestamp;
+        this.timestamp = new Date(newTimestamp.getTime());
```
Comment (tkaemming) [inline raven/src/main/java/com/getsentry/raven/event/BreadcrumbBuilder.java]: What's the rationale behind this change?

**Comment by tkaemming** [comment]
Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/raven-java/245)*

**Comment by bretthoerner** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[raven/src/main/java/com/getsentry/raven/event/BreadcrumbBuilder.java, line 36 \[r1\]](https://reviewable.io:443/reviews/getsentry/raven-java/245#-KPYXhQCFT6WC-kXyHTX-r1-36:-KPZFixRHNCR0ck-Mngp:b7gs5b) ([raw file](https://github.com/getsentry/raven-java/blob/533f450b5a1d26ff4cf25670233d99d787dea378/raven/src/main/java/com/getsentry/raven/event/BreadcrumbBuilder.java#L36)):*
<details><summary><i>Previously, tkaemming (ted kaemming) wrote…</i></summary>
> What's the rationale behind this change?
</details>
It warned that we're storing a mutable value. It must specifically look for `Date` (and others), because surely we do that often. But I figured there could be odd scenarios where people do something silly with breadcrumbs in a loop, reusing a single `Date`. It's pretty edge casey, I was thinking of mutable references (e.g. replacing the Date in a loop) which wouldn't actually be a problem... so idk.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/raven-java/245)*

**Comment by tkaemming** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[raven/src/main/java/com/getsentry/raven/event/BreadcrumbBuilder.java, line 36 \[r1\]](https://reviewable.io:443/reviews/getsentry/raven-java/245#-KPYXhQCFT6WC-kXyHTX-r1-36:-KPZY4Qpn4dOacDr188H:b-1xf9ie) ([raw file](https://github.com/getsentry/raven-java/blob/533f450b5a1d26ff4cf25670233d99d787dea378/raven/src/main/java/com/getsentry/raven/event/BreadcrumbBuilder.java#L36)):*
<details><summary><i>Previously, bretthoerner (Brett Hoerner) wrote…</i></summary>
> It warned that we're storing a mutable value. It must specifically look for `Date` (and others), because surely we do that often. But I figured there could be odd scenarios where people do something silly with breadcrumbs in a loop, reusing a single `Date`. It's pretty edge casey, I was thinking of mutable references (e.g. replacing the Date in a loop) which wouldn't actually be a problem... so idk.
</details>
Ah, okay.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/raven-java/245)*
